packages/utilities/src/IComponentAs.ts

```diff
@@ -4,4 +4,6 @@
  * @public
  */
 // export type IComponentAs<P> = string | ((props?: P) => JSX.Element);
-export type IComponentAs<T> = React.StatelessComponent<T> | React.ComponentClass<T>;
+export type IComponentAs<T> =
+  | React.StatelessComponent<T & { defaultRender?: React.StatelessComponent<T> }>
```

Comment:
This should be:
```ts
export type IComponentAs<T> = React.ComponentType<T & { defaultRender?: React.ComponentType<T>; }>;
```

Reply:
oh React.ComponentType is an actual type... When you originally suggested this in the other review I thought you were using 'ComponentType' as a placeholder/example lol